spyder/widgets/shortcuts.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2009- The Spyder Development Team
+# Licensed under the terms of the MIT License
+# (see spyderlib/__init__.py for details)
+
+"""Shortcut Summary dialog"""
+
+# TODO:
+# calculate windows size from screen size
+
+# Standard library imports
+from operator import itemgetter
+from itertools import groupby
+
+# Third party imports
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QFont
+from qtpy.QtWidgets import (QDialog, QLabel, QGridLayout, QGroupBox,
+                            QVBoxLayout, QHBoxLayout, QDesktopWidget)
+
+# Local imports
+from spyder.config.base import _
+from spyder.config.gui import iter_shortcuts
+
+# Constants
+MAX_FONT_SIZE = 16
+MIN_FONT_SIZE = 8
+
+class ShortCutsSummaryDialog(QDialog):
+    def __init__(self, parent=None, offset=0, force_float=False):
+        QDialog.__init__(self, parent=parent)
+
+        self._shortcuts_summary_title = _("Spyder Keyboard ShortCuts")
+
+        # Calculate font and amount of elements in each column according screen size
+        width, height = self.get_screen_resolution()
+        font_size = height / 80
+        font_size = max(min(font_size, MAX_FONT_SIZE), MIN_FONT_SIZE)
+        shortcuts_column = (height - 8 * font_size) / (font_size +16)
```

Comment:
This computation results in a window that is too tall on my laptop screen, which has:
```
width = 1366 
height = 737
font_size = 9.2125 
shortcuts_column = 26.30837878036688
```
The resulting window has height 771 (and width 1157).